data/tutorials/language/0lg_09_env.md

```diff
@@ -0,0 +1,558 @@
+---
+id: values-and-functions
+title: Values and Functions
+description: |
+  TBD
+category: "Language"
+---
+
+# Values and Functions
+
+## Introduction
+
+### Learning Goals
+
+When presenting OCaml or another functional programming language, it is often said: “Functions are treated as first-class citizens.” Without further explanation or context, this may not be helpful (it wasn't to me). The goal of this tutorial is to acquire the capabilities implied and entailed by this assertion. In turn, this should explain it:
+- Write all kinds of definitions, for all kinds of values, in all different ways
+- Accurately assess the scope of any definition
+- Avoid being fooled by shadowed definitions and scopes
+- Explain what a closure is and why it is so named
+- Partially apply functions, create closures, and pass them as values
+- Curry and uncurry functions
+- Use the `unit` type as a function parameter or return value, when needed
+
+### Chatting With OCaml
+
+This tutorial is shown using the UTop toplevel. Although other toplevels may be used, we recommend UTop. Code samples are intended to be copied/pasted into UTop. It would benefit the reader to write variations around the examples provided to strengthen understanding. The topics discussed are not limited to interactive execution of OCaml expressions. However, we believe they are easier to understand within the dynamics of interaction with the OCaml interpreter.
```

Comment:
```suggestion
This tutorial is shown using the UTop toplevel, which can be initiated with the `utop` command. Although other toplevels may be used, we recommend UTop. Code samples are intended to be copied/pasted into UTop. It would benefit the reader to write variations around the examples provided to strengthen understanding. The topics discussed are not limited to interactive execution of OCaml expressions. However, we believe they are easier to understand within the dynamics of interaction with the OCaml interpreter.
```